curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -76,10 +74,270 @@ CURVEFS_ERROR FuseS3Client::Init(const FuseClientOption &option) {
+                warmUpFile.fileLen, 0, &fi, data.get(), &rSize);
+            std::string file = data.get();
+            VLOG(9) << "file is: " << file;
+            // remove enter, newline, blank
+            std::string blanks("\r\n ");
+            file.erase(0, file.find_first_not_of(blanks));
+            file.erase(file.find_last_not_of(blanks) + 1);
+            VLOG(9) << "after eraser, file is: " << file;
+            std::list<std::string> splitPath;
+            std::string pDelimiter = "\n";
+            char* pToken = nullptr;
+            char* pSave = nullptr;
+            pToken = strtok_r(const_cast<char*>(file.c_str()),
+              const_cast<char*>(pDelimiter.c_str()), &pSave);
+            if (nullptr == pToken) {
+                VLOG(9) << "fetch end";
+                continue;
+           }
+            VLOG(9) << "pToken is: " << pToken
+                        << "pSave: " << pSave;
+            splitPath.push_back(pToken);
+            while (true) {
+                pToken = strtok_r(NULL,
+                  const_cast<char*>(pDelimiter.c_str()), &pSave);
+                if (nullptr == pToken) {
+                    VLOG(9) << "fetch end";
+                    break;
+                }
+                VLOG(9) << "pToken is: " << pToken
+                        << "pSave: " << pSave;
+                splitPath.push_back(pToken);
+            }
+            for (auto filePath : splitPath) {
+                FetchDentryEnqueue(filePath);
+            }
+        }
+        {   // file need warmup
+            std::list<fuse_ino_t> readAheadFiles;
+            readAheadFiles.swap(GetReadAheadFiles());
+            for (auto iter : readAheadFiles) {
+                VLOG(9) << "BackGroundFetch: " << iter;
+                fetchDataEnqueue(iter);
+            }
+        }
+        LOG_EVERY_N(WARNING, 100)
+            << "fetch thread end.";
+        usleep(WARMUP_CHECKINTERVAL_US);
+    }
+    return;
+}
+
+void FuseS3Client::fetchDataEnqueue(fuse_ino_t ino) {
+    VLOG(9) << "fetchDataEnqueue start: " << ino;
+    auto task = [this, ino]() {
+        std::shared_ptr<InodeWrapper> inodeWrapper;
+        CURVEFS_ERROR ret = inodeManager_->GetInode(ino, inodeWrapper);
+        if (ret != CURVEFS_ERROR::OK) {
+            LOG(ERROR) << "inodeManager get inode fail, ret = " << ret
+                    << ", inodeid = " << ino;
+            return;
+        }
+        google::protobuf::Map<uint64_t, S3ChunkInfoList> *s3ChunkInfoMap
+          = nullptr;
+        {
+            ::curve::common::UniqueLock lgGuard = inodeWrapper->GetUniqueLock();
+            s3ChunkInfoMap = inodeWrapper->GetChunkInfoMap();
+        }
+        if (nullptr == s3ChunkInfoMap ||
+          s3ChunkInfoMap->empty()) {
+            return;
+        }
+        travelChunks(ino, s3ChunkInfoMap);
+    };
+    GetTaskFetchPool().Enqueue(task);
+}
+
+void FuseS3Client::travelChunk(fuse_ino_t ino, S3ChunkInfoList chunkInfo,
```

Comment:
What is this function for?

Reply:
travel and download all objs belong to the chunk. and i have commnted.